packages/flutter_tools/templates/app_shared/windows.tmpl/runner/Runner.rc.tmpl

```diff
@@ -60,21 +60,31 @@ IDI_APP_ICON            ICON                    "resources\\app_icon.ico"
 // Version
 //
 
-#ifdef FLUTTER_BUILD_NUMBER
-#define VERSION_AS_NUMBER FLUTTER_BUILD_NUMBER
+// The extra level of indirection is necessary to make the preprocessor expand
```

Comment:
I don't think we can plausibly validate 2; it could be set in the project itself, or in an included file.

I think the sane options would be:
1. Just do the migration if it doesn't conflict. For some people things will just start working (the updated PR won't conflict with nearly as many people, so it could even be most people), for some it'll be a no-op because they've replaced the use of the variables with hard-coded values, and for a very small number who plumbed this into their build system themselves it'll replace their workaround with something that works automatically (which seems pretty low-risk to me).
2. Don't ever try to auto-migrate; instead just warn if the new code can't be found and suggest they update manually (with instructions).

Reply:
Here is the design doc on the upcoming `flutter migrate` command: flutter.dev/go/flutter-migrate

From my understanding, `flutter migrate` does option #2: `flutter upgrade` and `flutter doctor` will warn if your project needs to be updated, and the user will need to run `flutter migrate` to apply the migration.

Reply:
Ah that makes sense; at least this is a user-initiated action, so assuming it emits appropriate messages informing the user of what we're about to attempt and whether or not it was successful, it should be pretty unsurprising to the user. If they've got their app under source control (and hopefully they do), they can inspect the diffs themselves after the action is complete and elect not to commit if they don't like what they see.